PSKoans/Public/Show-Karma.ps1

```diff
@@ -80,6 +85,30 @@ function Show-Karma {
                 $KoanLocation | Invoke-Item
             }
         }
+        'OpenFile' {
+            try {
+                $Results = Get-Karma @GetParams
+            }
+            catch {
+                $PSCmdlet.ThrowTerminatingError($_)
+            }
+            $UnformatedPath     = $Results.Meditation
+            $PSKoansFilePath = $UnformatedPath.Substring($UnformatedPath.IndexOf(', ')+2,$UnformatedPath.IndexOf(': ')-($UnformatedPath.IndexOf(', ')+2))
+            $PSKoansLine = $UnformatedPath.Substring($UnformatedPath.IndexOf(': line ')+7,$UnformatedPath.LastIndexOf(("`n"))-($UnformatedPath.IndexOf(': line ')+7))    
```

Comment:
You're right, there has to be a better way to figure out the current topic path! Not sure about how to jump to the exact line number, that might require something more complex...

Anyway, you can get the file path much easier by doing something like this:

```powershell
$Results = Get-Karma @GetParams

$Path = Get-PSKoan -Topic $Results.CurrentTopic.Name -Scope User |
    Select-Object -ExpandProperty Path
```

As for the line number... since the meditation message is split over two lines always, you can grab the second line and remove everything after the colon:

```powershell
($Results.Meditation -split '\r?\n')[1] -replace ':.+'
```

Reply:
Make sure you add comments to the latter one, it's a bit of a mouthful of code, in the end. 

Perhaps we could also alter `Get-Karma` to add the line number and full topic path to the `CurrentTopic` properties, making this part trivial? The information `Get-Karma` has access to should allow us to get the line number a bit more easily, I think, and it definitely has the path information available, or we could just get it the same way and store it from the beginning.

That would also make it easier for other folks to write their own Show-Karma style wrappers that have similar capabilities.